src/mctpd.c

```diff
@@ -4526,7 +4615,42 @@ static int add_interface(struct ctx *ctx, int ifindex)
 	}
 
 	if (phys_binding == MCTP_PHYS_BINDING_PCIE_VDM) {
-		link->discovered = DISCOVERY_UNDISCOVERED;
+		link->discovery.flag = DISCOVERY_UNDISCOVERED;
```

Comment:
We want the discovery process (below) to work for any binding type that supports Discovery Notify, not just VDM, right?

Should we move that out to be conditional on `link->discovery.flag == DISCOVERY_UNDISCOVERED` ?

(there may be some binding-specific addressing info needed, but the rest should be common?)